frontend/lib/framework/ID.cpp

```diff
@@ -129,6 +129,20 @@ ID ID::fabricateId(Context* context,
   return newId;
 }
 
+ID ID::generatedId(UniqueString symbolPath,
+                 int postOrderId,
+                 int numChildIds) {
+  int pid;
+  if (postOrderId == -1) {
+    pid = ID_GEN_START;
+  } else {
+    pid = ID_GEN_START - 1 - postOrderId;
```

Comment:
Could you add a check that we're packing and unpacking these correctly in to `testIds.cpp` ? by calling `generatedId` and checking that `postOrderId` comes out right?

Reply:
Great suggestion! I had introduced a bug in a previous PR that is now fixed.